app/javascript/components/Schedule.js

```diff
@@ -100,6 +100,66 @@ class Schedule extends Component {
     })
   }
 
+  findTimeSlotConflicts = (previewSlots) => {
+    const { slots } = this.state
+    let conflicts = []
+
+    previewSlots.forEach(ps => {
+      slots.forEach(s => {
+        let sameDaySameRoom = s.room_id === parseInt(ps.room) && s.conference_day === ps.day
+
+        if (sameDaySameRoom) {
+          let timeConflict = ps.startTime > this.ripTime(s.start_time) && ps.startTime < this.ripTime(s.end_time) || ps.endTime > this.ripTime(s.start_time) && ps.startTime < this.ripTime(s.end_time)
```

Comment:
@YayFiber I see a possible refactoring of logic here to a function.

Reply:
Good call, I will add this today.